glean-core/src/ping/mod.rs

```diff
@@ -175,10 +175,38 @@ impl PingMaker {
         if let Some(client_info) =
             StorageManager.snapshot_as_json(glean.storage(), "glean_client_info", true)
         {
+            log::error!("db glean_client_info: {:?}", client_info);
             let client_info_obj = client_info.as_object().unwrap(); // safe unwrap, snapshot always returns an object.
-            for (_key, value) in client_info_obj {
-                merge(&mut map, value);
+            for (_metric_type, metrics) in client_info_obj {
+                merge(&mut map, metrics);
             }
+            log::error!("flattened client_info: {:?}", map);
+            let map = map.as_object_mut().unwrap(); // safe unwrap, we created the object above.
+            let mut attribution = serde_json::Map::new();
+            let mut distribution = serde_json::Map::new();
+            map.retain(|name, value| {
+                let split: Vec<_> = name.split('.').collect();
```

Comment:
I stumbled here at first.
Technically the category can contain dots.
I guess here it is correct because we control all metrics in client_info and we don't have dotted categories.

But do we need to collect? That definitely allocates. If we just make use of the iterator (`split.next()`) we could avoid that.

Reply:
I wanted a nice way to name the parts of the split, but you're right. I'll work around it.